rust/sbp2json/src/lib.rs

```diff
@@ -99,16 +99,9 @@ where
     I: Iterator<Item = std::result::Result<M, E>> + 'a,
     E: std::error::Error + 'a,
 {
-    if fatal_errors {
-        Box::new(messages.take_while(|m| m.is_ok()).map(|m| m.unwrap()))
-    } else {
-        Box::new(messages.filter_map(|m| {
-            if let Ok(m) = m {
-                Some(m)
-            } else {
-                eprintln!("{:?}", m.err());
-                None
-            }
-        }))
-    }
+    let log_err = move |e| match fatal_errors {
+        true => panic!("{:?}", e),
```

Comment:
Not sure we should panic, this is a pretty big behavior change, with fatal_errors I'm assuming you could catch the errors further up the chain, without "fatal errors" we just swallow errors?

Reply:
yea think --fatal-errors just makes it so it stops after error,
which more weird is not sure if this was just to fix the case of unterminating iterator because looks like running json2sbp on a broken json will not terminate

Reply:
maybe we just have --fatal-errors not log error ? since we take until OK, so we bascially assume the next one must be a parse error anyways